controllers/vspherecluster_reconciler.go

```diff
@@ -117,18 +118,90 @@ func (r *clusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return r.reconcileDelete(ctx, clusterContext)
 	}
 
-	if cluster == nil {
-		log.Info("Waiting for Cluster controller to set OwnerRef on VSphereCluster")
-		return reconcile.Result{}, nil
-	}
-
 	// Handle non-deleted clusters
 	return r.reconcileNormal(ctx, clusterContext)
 }
 
+// patch updates the VSphereCluster and its status on the API server.
+func (r *clusterReconciler) patch(ctx context.Context, clusterCtx *capvcontext.ClusterContext) error {
+	// always update the readyCondition.
+	conditions.SetSummary(clusterCtx.VSphereCluster,
+		conditions.WithConditions(
+			infrav1.VCenterAvailableCondition,
+		),
+	)
+
+	if !v1beta2conditions.Has(clusterCtx.VSphereCluster, infrav1.VSphereClusterVCenterAvailableV1Beta2Condition) {
```

Comment:
I'm wondering a little bit about this.
If the goal is to handle code paths where those conditions are not set after an upgrade, this could lead to a state where the condition is false and it is never moved to true, which is an issue.

What if instead, if the condition is missing, we set this condition to true if the vsphere cluster is ready, false otherwise

Reply:
ack, misinterpreted what was discussed :-)

Adjusting it to set them on ignore. The old summary only included VCenterAvailable, so we also will rely on that, and add the others with ignore.